index.d.ts

```diff
@@ -217,6 +217,22 @@ declare module '@primer/components' {
 
   export const Link: React.FunctionComponent<LinkProps>
 
+  export type PaginationHrefBuilder = (page: number) => string
+
+  export type PaginationPageChangeCallback = (e: React.MouseEvent, page: number) => void
+
+  export interface PaginationProps extends CommonProps {
+    currentPage: number
+    hrefBuilder?: PaginationHrefBuilder
+    marginPageCount?: number
+    onPageChange?: PaginationPageChangeCallback
+    pageCount: number
+    showPages?: boolean
+    surroundingPageCount?: number
```

Comment:
Adding doc comments on some fields here (`/***` should autogenerate it in VS Code) might help. At first glance I'm not sure what `surroundingPageCount` and `marginPageCount` do or how they're different.